src/ape_ethereum/ecosystem.py

```diff
@@ -75,7 +92,83 @@ def encode(self) -> bytes:
         return signed_txn
 
 
+class StaticFeeTransaction(BaseTransaction):
+    """
+    Transactions that are pre-EIP-1559 and use the ``gasPrice`` field.
+    """
+
+    gas_price: int = None  # type: ignore
+    type: str = "0x0"
+
+    @property
+    def max_fee(self) -> int:
+        return (self.gas_limit or 0) * (self.gas_price or 0)
+
+    def set_defaults(self, provider: ProviderAPI):
+        if self.gas_price is None:
+            self.gas_price = provider.gas_price
```

Comment:
This would query the value of gas price only at this point in time. Perhaps it should be more lazily querying.

Reply:
This only gets called right before sending a transaction in `AccountAPI.call()`.
Would a warning suffice for this method? Like, `Only set the defaults on the transaction before sending it otherwise they may become outdated."` ?

Otherwise, the transaction almost need a provider as its dependency (I guess it kind of does in this sense).